.github/workflows/unittests.yml

```diff
@@ -0,0 +1,129 @@
+name: Unittests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+permissions:
+  actions: write  # Needed for skip-duplicate-jobs job
+  contents: read
+  checks: write   # Needed for publish test results job
+  pull-requests: write   # Needed for publish test results job
+
+jobs:
+  # Special job which automatically cancels old runs for the same branch, prevents runs for the
+  # same file set which has already passed, etc.
+  pre_job:
+    name: Skip Duplicate Jobs Pre Job
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@9d116fa7e55f295019cfab7e3ab72b478bcf7fdd # v4.0.0
+        with:
+          cancel_others: 'true'
+          github_token: ${{ github.token }}
+
+  coverage:
+    runs-on: ubuntu-latest
+    if: success() || failure()
+    needs:
+      - pre_job
+      - unittests
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        id: setup-python
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install pycoverage
+        run: pip install coverage==4.5.4
+      - name: Download coverage reports
+        uses: actions/download-artifact@v3
+        with:
+          path: reports
+
+      - name: Prepare Coverage Data for codecov.io
+        run: |
+          coverage combine reports/**/.coverage
+          coverage xml -i -o coverage.xml
+
+      - name: Upload Coverage to Codecov.io
+        uses: codecov/codecov-action@v3
+        with:
+          files: ./coverage.xml
+          fail_ci_if_error: true
+
+  unittests:
+    name: Unittests - Python ${{ matrix.python-version }} - ${{ matrix.platform }}
+    runs-on: ${{ matrix.platform }}-latest
+    defaults:
+      run:
+        shell: bash
+    needs: pre_job
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - "3.6"
+          - "3.7"
+          - "3.8"
+          - "3.9"
+          - "3.10"
+        platform:
+          - ubuntu
+          - macos
+        include:
+          - platform: windows
+            python-version: "3.10"
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Setup Python
```

Comment:
To speed up the build, I assume we also want to add step where we cache the pip dependencies.

Reply:
Not really needed ATM. It pretty much is doing that already for the requirements I'm installing from the commandline. That pip install step is sub-10sec.
Right now the docker image tests take more time than these unittests. We can revisit caching if it becomes problematic.

Reply:
With pip dependency caching enabled, that cache should also be used by tox, so it should also speed tox target run